### PR TITLE
Add release milestone check

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area dev-productivity
/kind enhancement

**What this PR does / why we need it**:
With this PR, when a milestone for the next minor version exists, PRs are only mergeable if they are assigned to this milestone. Otherwise, the merge will be blocked.
This will help the developers who are currently in the release validation phase to prevent unexpected PR merges to `master`.

**Special notes for your reviewer**:
/cc @ialidzhikov @timebertt 

**Release note**:

```noteworthy developer
If a milestone for the next minor version exists then PRs to the `master` branch are only mergeable if they are assigned to this milestone.
```

### CODE REVIEW

File: .github/workflows/release-milestone.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+          for (const milestone of milestones.data) {
+            if (milestone.title != nextMinorVersion) {
+              continue;
+            }
+
+            // milestone found, check if PR is associated
+            milestoneForNextMinorReleaseFound = true;
+
+            if (context.payload.pull_request.milestone == null) {
```
Comment (timebertt) [inline .github/workflows/release-milestone.yaml:56]: What about the case, that the PR is marked for a different milestone?

**Review by timebertt — COMMENTED** [review]
Very nice, love it!

File: .github/workflows/release-milestone.yaml (hunk 1 above)
Comment (rfranzke) [reply .github/workflows/release-milestone.yaml:56]: I also thought about it, but I think this case doesn't exist today in our workflow, so I went for ignoring it to keep the script simple. WDYT?

Comment (timebertt) [reply .github/workflows/release-milestone.yaml:56]: Fair enough :)

**Review by timebertt — APPROVED** [review]
/lgtm

**Pull request merged**